src/actions/ast.js

```diff
@@ -126,3 +138,64 @@ export function setSelection(
     });
   };
 }
+
+/**
+ * Action dispatched from stepOver action. Receives the position
+ * and depending on where it is paused, sets up hidden breakpoints
+ * and handles resume, stepOver (as needed)
+ * @param position
+ * @returns {function(ThunkArgs)}
+ */
+export function analyzeStepping(type) {
+  return async ({ dispatch, getState, client }: ThunkArgs) => {
+    const position = await dispatch(getPosition());
```

Comment:
if we don't intend to update the state, it shouldn't be an action. We can just as easily pass in `getState()`